src/main/java/io/servertap/api/v1/PlayerApi.java

```diff
@@ -25,11 +34,89 @@ public static void playersGet(Context ctx) {
             p.setWhitelisted(player.isWhitelisted());
             p.setBanned(player.isBanned());
             p.setOp(player.isOp());
+            //getBalance(String) is deprecated it is now encouraged to use getBalance(Offline player)
 
+            double curbalance = PluginEntrypoint.getEconomy().getBalance(player);
+            p.setBalance(curbalance);
             players.add(p);
         }));
 
         ctx.json(players);
     }
 
+    public static void playerGet(Context ctx) {
+        Player p = new Player();
+
+        org.bukkit.entity.Player player= Bukkit.getPlayer(ctx.pathParam("player"));
+
+
+        double curbalance = PluginEntrypoint.getEconomy().getBalance(player);
+        p.setBalance(curbalance);
+        p.setUuid(player.getUniqueId().toString());
+        p.setDisplayName(player.getDisplayName());
+
+        p.setAddress(player.getAddress().getHostName());
+        p.setPort(player.getAddress().getPort());
+
+        p.setExhaustion(player.getExhaustion());
+        p.setExp(player.getExp());
+
+        p.setWhitelisted(player.isWhitelisted());
+        p.setBanned(player.isBanned());
+        p.setOp(player.isOp());
+
+
+        ctx.json(p);
+    }
+
+
+
+    public static void offlinePlayersGet(Context ctx) {
+
+        ArrayList<io.servertap.api.v1.models.OfflinePlayer> players = new ArrayList<>();
+
+
+        OfflinePlayer[] offlinePlayers = Bukkit.getOfflinePlayers();
+
+        for (int i = 0; i <offlinePlayers.length ; i++) {
+            io.servertap.api.v1.models.OfflinePlayer p = new io.servertap.api.v1.models.OfflinePlayer();
+            OfflinePlayer player=offlinePlayers[i];
+
+            p.setDisplayName(player.getName());
+            p.setUuid(player.getUniqueId().toString());
+            p.setWhitelisted(player.isWhitelisted());
+            p.setBanned(player.isBanned());
+            p.setOp(player.isOp());
+
+            double curbalance = PluginEntrypoint.getEconomy().getBalance(offlinePlayers[i]);
+
+            p.setBalance(curbalance);
+            players.add(p);
+
+        }
+
+        ctx.json(players);
+    }
+
+
+
+
+    public static void playerPay(Context ctx) {
```

Comment:
In future we should break this out into a whole `EconomyApi` class I think. For now it's fine here

Reply:
I'll get on that

Reply:
Awesome. Maybe move the route under `/economy` as well, as per #10